### ISSUE TITLE
Managed Items are not removed from deleted group

### ISSUE THREAD

**Issue by falmanna** [openhab/openhab-core#1392]
I noticed that when I deleted a managed group item using the paperui, its children are not removed automatically from the group. Shouldn't this be the correct behavior for managed items?

**Comment by cweitkamp** [comment]
Can you check if the same behavior exists if you try to remove a group item via [console](https://www.openhab.org/docs/administration/runtime.html#items) command: `smarthome:items remove <itemname>`?

**Comment by falmanna** [comment]
Yes, the same behavior exists.

**Comment by falmanna** [comment]
After digging into the code for a few days, I found that there is no handling for the deletion of a group. So I added this piece and it is working with me.

```
public class ItemRegistryImpl extends AbstractRegistry<Item, String, ItemProvider> implements ItemRegistry {
.
.
.
    //remove group names from mombers
    private void removeMembersFromGroupItem(GroupItem groupItem) {
        groupItem.getMembers().forEach(memberItem -> {
            if (!(memberItem instanceof GenericItem)) {
                return;
            }
            GenericItem genericMemberItem = (GenericItem) memberItem;
            genericMemberItem.removeGroupName(groupItem.getName());
        });
    }

    @Override
    protected void onRemoveElement(Item element) {
        //If it is a group item, remove members before disposing
        if (element instanceof GroupItem) {
            removeMembersFromGroupItem((GroupItem) element);
        }
        if (element instanceof GenericItem) {
            ((GenericItem) element).dispose();
        }
        removeFromGroupItems(element, element.getGroupNames());
    }
.
.
.
}
```

**Comment by Rossko57** [comment]
I'm not sure if that is desired behaviour.  Isn't it legitimate for an Item to belong to a non-existent Group? i.e. you might not have defined it yet.   I appreciate that in this case you are deleting the Group, but not entirely convinced that should go around and edit member Items.

What happens to members if you edit an existing Group Item?

**Comment by falmanna** [comment]
> What happens to members if you edit an existing Group Item?

What do you mean?

**Comment by Rossko57** [comment]
Sorry, I will expand,
Editing the properties of some Group Item could be like deleting/recreating it - I don't know the implementation details, there is more than one way to edit, which may matter.

I was worrying the editing may accidentally trigger a clean-up of member Items,

**Comment by mjcumming** [comment]
I think this would be okay if its deliberate user action from the PaperUI. It would cause problems with text configurations - ie if there was an error in a .items file and you save it, then any groups get removed from OH and then items that were members of those groups have their membership removed which is not desirable.

**Comment by Rossko57** [comment]
Understood I think - accidentally invalidating an xxx.items file during editing would uhh destroy the group instead of keeping the already-in-use version.
Does not sound like a terrible handicap to me, fixing the xxx.items file should restore all?  Including say members defined in other files, or is that the problem you see?

**Comment by J-N-K** [comment]
I think @mjcumming wants to point out that after automatically removing managed items from the deleted group leads to a situation where - after the file was fixed and the group is back - the managed items are missing from that group. 

IMO it would be better to provide a manual "clean-up" like I implemented for links in #1395.
